torch/utils/_cuda_trace.py

```diff
@@ -0,0 +1,75 @@
+import traceback
+import sys
+from typing import Callable, List, Tuple, Any
+import logging
+
+
+class CallbackRegistry:
+    def __init__(self, name: str):
+        self.name = name
+        self.callback_list: List[Callable[[Tuple[Any, ...]], None]] = []
+
+    def add_callback(self, cb: Callable[[Tuple[Any, ...]], None]) -> None:
+        self.callback_list.append(cb)
+
+    def fire_callbacks(self, *args: int) -> None:
+        logger = logging.getLogger(__name__)
+        for cb in self.callback_list:
+            try:
+                cb(*args)
+            except Exception as e:
+                logger.exception(
+                    f"Exception in callback for {self.name} registered with CUDA trace",
+                    e,
+                )
+
+
+CUDAEventCreationCallbacks = CallbackRegistry("CUDAEventCreation")
+CUDAEventDeletionCallbacks = CallbackRegistry("CUDAEventDeletion")
+CUDAEventRecordCallbacks = CallbackRegistry("CUDAEventRecord")
+CUDAEventWaitCallbacks = CallbackRegistry("CUDAEventWait")
+CUDAMemoryAllocationCallbacks = CallbackRegistry("CUDAMemoryAllocation")
+CUDAMemoryDeallocationCallbacks = CallbackRegistry("CUDAMemoryDeallocation")
+CUDAStreamCreationCallbacks = CallbackRegistry("CUDAStreamCreation")
+
+
+def register_callback_for_cuda_event_creation(
+    cb: Callable[[Tuple[Any, ...]], None]
```

Comment:
Could we make this type annotation be made more precise?

Reply:
Done!